config/action.d/badips.py

```diff
@@ -80,13 +82,14 @@ class BadIPsAction(ActionBase):
 	"""
 
 	_badips = "http://www.badips.com"
-	_Request = partial(
-		Request, headers={'User-Agent': "Fail2Ban %s" % f2bVersion})
+	def _Request(self, url, **argv):
+		return Request(url, headers={'User-Agent': self.agent}, **argv)
 
 	def __init__(self, jail, name, category, score=3, age="24h", key=None,
-		banaction=None, bancategory=None, bankey=None, updateperiod=900):
+		banaction=None, bancategory=None, bankey=None, updateperiod=900, agent=None):
 		super(BadIPsAction, self).__init__(jail, name)
 
+		self.agent = agent if agent is not None else ("Fail2Ban/%s" % f2bVersion)
```

Comment:
Which reveals the duplication... Why not to define fail2ban_agent along with the version, to be also reused here?